feature: counter increase function (#45)
mondaycom/apps-sdk: README.md

</details>

## Operations counter

<details>

> <summary>A tool that provide the ability to count the number of operations base on a pre-defined period.</summary>

Reviewer: ```suggestion
<summary>A tool that provides the ability to count the number of operations base on a pre-defined period.</summary>
```

- This is the way to count operations around your project.
- Has 3 pre-defined periods
  - `DAILY` - the counter will be reset every day (base on GMT time zone)
  - `MONTHLY` - the counter will be reset every month (base on GMT time zone).

>   - `YEARLY` -  the counter will be reset every year (base on GMT time zone).

Reviewer: What does it mean `base on GMT time zone`?

Author: @gregra81 
Greenwich Mean Time (GMT) 
universal time
GMT 0
https://www.google.com/search?q=gmt+time+now&oq=hgmt&gs_lcrp=EgZjaHJvbWUqDwgEEAAYChiLAxixAxiABDIGCAAQRRg5MhIIARAAGAoYgwEYsQMYyQMYgAQyDwgCEAAYChiDARixAxiABDIMCAMQABgKGIsDGIAEMg8IBBAAGAoYiwMYsQMYgAQyEggFEAAYChiDARiLAxixAxiABDIJCAYQABgKGIAEMgkIBxAAGAoYgAQyCQgIEAAYChiABDIMCAkQABgKGJIDGIAE0gEINTIzNGoxajGoAgCwAgA&sourceid=chrome&ie=UTF-8

https://www.google.com/search?q=universal+time+now+VS+GMT&sca_esv=589592780&ei=JfR1ZfeQHp-lkdUPzcOcqAs&ved=0ahUKEwi3t4SlsYWDAxWfUqQEHc0hB7UQ4dUDCBA&uact=5&oq=universal+time+now+VS+GMT&gs_lp=Egxnd3Mtd2l6LXNlcnAiGXVuaXZlcnNhbCB0aW1lIG5vdyBWUyBHTVQyBRAhGKABMgUQIRigATIFECEYoAEyBRAhGKABMggQIRgWGB4YHUjpQ1CwFlibL3AHeAGQAQGYAc4DoAGAEaoBCTAuNC4yLjIuMbgBA8gBAPgBAcICChAAGEcY1gQYsAPCAg0QABiABBiKBRhDGLADwgIFEAAYgATCAgYQABgWGB7CAgoQIRgWGB4YDxgdwgIEECEYFeIDBBgAIEGIBgGQBgo&sclient=gws-wiz-serp

Reviewer: ```
  - `DAILY` - the counter will be reset every day (at 00:00 GMT)
  - `MONTHLY` - the counter will be reset every month
  - `YEARLY` -  the counter will be reset every year
```

> - You can increase the counter by a positive number e.g. if the current counter value is 4, and you like a specific operation to increase to 6 than you add 2 instead of the default which is 1, you can do that by using the BODY `incrementBy` property.
> - You can generate on the fly new counters e.g. for DOC caning and a different counter for IMAGE scanning by using the BODY `kind` property.

Reviewer: ```suggestion
- You can increase the counter any positive number. By default it increases by 1
- We support different counter kinds, with a default counter that does not have any kind and is meant to simply count the usage of your workflow / app. If you want to count multiple types of operations e.g. for DOC scanning and  for IMAGE scanning you can supply the optional `kind` property.
```

#### initialize

```typescript
import { Storage } from '@mondaycom/apps-sdk';
const storage = new Storage('<ACCESS_TOKEN>');
```

#### incrementCounter

- `Period: string` - This is an enum for those value: `DAILY`, `MONTHLY`, `YEARLY`

> - `incrementBy?: number` - by how much to increase the current counter value (_OPTIONAL_)

Reviewer: ```suggestion
- `incrementBy?: number` - by how much to increase the current counter value (_OPTIONAL_). Default is 1
```

- `kind?: string` - set a new counter for the `kind` operation name (_OPTIONAL_)

```typescript
const { message, newCounterValue, success, error } = await storage.incrementCounter(Period.MONTHLY, {
  incrementBy: 2,

>   kind: 'foo'

Reviewer: ```suggestion
  kind: 'image.scan'
```

Author: I limited the special chars to -_
"." is not included, So I set it to 

kind: 'image_scan'
